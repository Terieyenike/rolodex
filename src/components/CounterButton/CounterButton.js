import React from 'react';

class CounterButton extends React.Component {
  state = {
    count: 0,
  };

  shouldComponentUpdate(nextProps, nextState) {
    if (this.state.count !== nextState.count) {
      return true;
    }
    return false;
  }

  render() {
    return (
      <button
        id='counter'
        onClick={() => this.setState((state) => ({ count: state.count + 1 }))}
        color={this.props.color}
      >
        Count: {this.state.count}
      </button>
    );
  }
}

export default CounterButton;
