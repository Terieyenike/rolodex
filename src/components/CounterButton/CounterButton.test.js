import React from 'react';
import { shallow } from 'enzyme';
import CounterButton from './CounterButton';

describe('CounterButton', () => {
  it('expect to render the Card component', () => {
    const mockColor = 'red';
    expect.assertions(1);
    expect(shallow(<CounterButton color={mockColor} />)).toMatchSnapshot();
  });

  it('renders without crashing', () => {
    expect(shallow(<CounterButton />)).toMatchSnapshot();
  });

  it('correctly increment the counter', () => {
    const wrapper = shallow(<CounterButton />);
    expect(wrapper).toMatchSnapshot();
    wrapper.find('[id="counter"]').simulate('click');
    expect(wrapper.state()).toEqual({ count: 1 });
    wrapper.find('[id="counter"]').simulate('click');
    wrapper.find('[id="counter"]').simulate('click');
    expect(wrapper.state()).toEqual({ count: 3 });
  });
});
