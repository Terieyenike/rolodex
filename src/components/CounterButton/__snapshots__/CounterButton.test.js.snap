// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CounterButton correctly increment the counter 1`] = `ShallowWrapper {}`;

exports[`CounterButton expect to render the Card component 1`] = `ShallowWrapper {}`;

exports[`CounterButton renders without crashing 1`] = `ShallowWrapper {}`;
